src/software/backend/simulation/physics/physics_robot.cpp

```diff
@@ -24,119 +63,406 @@ PhysicsRobot::~PhysicsRobot()
+    double total_shape_area = 0.0;
+    for (const auto* shape : body_shapes)
+    {
+        total_shape_area += polygonArea(*shape);
+    }
+    robot_body_fixture_def.density = mass_kg / total_shape_area;
+
+    for (const auto shape : body_shapes)
+    {
+        robot_body_fixture_def.shape = shape;
+        robot_body->CreateFixture(&robot_body_fixture_def);
+    }
+}
+
+void PhysicsRobot::setupDribblerFixture(const Robot& robot, double dribbler_depth)
+{
+    b2FixtureDef robot_dribbler_fixture_def;
+    // The dribbler has no mass in simulation. Mass is already accounted for by the robot
+    // body
+    robot_dribbler_fixture_def.density = 0.0;
+    // We explicitly choose to make the dribbler NOT a sensor, because sensor fixtures do
+    // not trigger PreSolve contact callbacks, which we rely on to apply dribbling force
+    // at every physics step
+    robot_dribbler_fixture_def.isSensor = false;
+    // We assume the ContactListener for the world will disable collisions with the
+    // dribbler so the restitution and friction don't matter
+    robot_dribbler_fixture_def.restitution = 0.0;
+    robot_dribbler_fixture_def.friction    = 0.0;
+    robot_dribbler_fixture_def.userData =
+        new PhysicsObjectUserData({PhysicsObjectType::ROBOT_DRIBBLER, this});
+
+    // Box2D requires that polygon vertices are specified in counter-clockwise order.
+    // The fixture shape is added relative to the body in its local coordinate frame,
+    // so we do not need to rotate the points to match the orientation of the robot.
+    const unsigned int num_vertices              = 4;
+    b2Vec2 dribbler_shape_vertices[num_vertices] = {
+        createVec2(Point(DIST_TO_FRONT_OF_ROBOT_METERS, DRIBBLER_WIDTH_METERS / 2.0)),
+        createVec2(Point(DIST_TO_FRONT_OF_ROBOT_METERS - dribbler_depth,
+                         DRIBBLER_WIDTH_METERS / 2.0)),
+        createVec2(Point(DIST_TO_FRONT_OF_ROBOT_METERS - dribbler_depth,
+                         -DRIBBLER_WIDTH_METERS / 2.0)),
+        createVec2(Point(DIST_TO_FRONT_OF_ROBOT_METERS, -DRIBBLER_WIDTH_METERS / 2.0))};
+    b2PolygonShape* dribbler_shape = new b2PolygonShape();
+    dribbler_shape->Set(dribbler_shape_vertices, num_vertices);
+    robot_dribbler_fixture_def.shape = dribbler_shape;
+    robot_body->CreateFixture(&robot_dribbler_fixture_def);
+}
+
+void PhysicsRobot::setupChickerFixture(const Robot& robot, double total_chicker_depth,
+                                       double chicker_thickness)
+{
+    b2FixtureDef robot_chicker_fixture_def;
+    // We want perfect damping to help us keep the ball when dribbling
+    robot_chicker_fixture_def.restitution = 0.0;
+    // We want lots of friction for when the ball is being dribbled so it stays controlled
+    robot_chicker_fixture_def.friction = 1.0;
+    // The chicker has no mass in simulation. Mass is already accounted for by the robot
+    // body
+    robot_chicker_fixture_def.density = 0.0;
```

Comment:
nit: can we put this number in the header file

Reply:
done